Sources/FirebaseFirestore/Firestore+Swift.swift

```diff
@@ -35,3 +35,93 @@ extension Firestore {
     swift_firebase.swift_cxx_shims.firebase.firestore.firestore_collection(self, std.string(collectionPath))
   }
 }
+
+// An extension that adds the encoder and decoder functions required
+// to serialize and deserialize documents in Firebase. These are mostly
+// copy of the Swift implementation that's available within firebase-ios-sdk
+//
+public extension Firestore {
+    class Encoder {
+      public var dateEncodingStrategy: FirebaseDataEncoder.DateEncodingStrategy = .timestamp
+      public var dataEncodingStrategy: FirebaseDataEncoder.DataEncodingStrategy = .blob
```

Comment:
Why are these `public`?

Reply:
This is the same API as in the Cocoa SDK, https://github.com/firebase/firebase-ios-sdk/blob/afca2896b91a5b5d06623e11351a0255a32983ff/Firestore/Swift/Source/Codable/EncoderDecoder.swift#L26 I'd prefer to keep these as a 1:1 match to ensure more integration compatibility.

Reply:
As talked about offline we're rebuilding the same implementation from the Cocoa APIs to ensure compatibility with our code and provide a familiar set of defaults for anyone else coming from the existing Cocoa SDK.